3rdParty/velocypack/include/velocypack/Iterator.h

```diff
@@ -34,9 +34,13 @@
 
 namespace arangodb::velocypack {
 
-class ArrayIterator : public std::iterator<std::forward_iterator_tag, Slice> {
+class ArrayIterator {
  public:
   using iterator_category = std::forward_iterator_tag;
+  using difference_type = std::ptrdiff_t;
+  using value_type = Slice;
+  using pointer = Slice*;
+  using reference = Slice&;
```

Comment:
does this need to be ported to velocypack main repo?

Reply:
Yes. `std::iterator` is deprecated since C++17, so this needs to be done sooner or later anyway.